### PR TITLE
Revert "Raise user handler exception in MultiThreadedExecutor. (#984)"

### PR DESCRIPTION
**Pull request by fujitatomoya**
This reverts commit 6c0847a70df2a359c28ffae7e49ece2e62c1794d.

Signed-off-by: Tomoya Fujita <Tomoya.Fujita@sony.com>

### CODE REVIEW

**Comment by fujitatomoya** [comment]
@ivanpauno @wjwwood i need to revert https://github.com/ros2/rclpy/pull/984 for now, see https://github.com/ros2/rclpy/issues/1016.

and then, i will come up with complete fix including https://github.com/ros2/rclpy/issues/1016.

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17439/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11964/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17949/)

**Comment by ivanpauno** [comment]
@fujitatomoya please in the future, consider backporting only after the issue has been available in rolling for some time (e.g. 3 or 6 months), so there's time to detect issues.

> @ivanpauno @wjwwood i need to revert https://github.com/ros2/rclpy/pull/984 for now, see https://github.com/ros2/rclpy/issues/1016.

Could you explain the issue in more detail?
I don't understand how reverting this is related to the issue described there.

**Comment by ivanpauno** [comment]
> @fujitatomoya please in the future, consider backporting only after the issue has been available in rolling for some time (e.g. 3 or 6 months), so there's time to detect issues.

@clalancette  do we have any official policy about this?

**Comment by clalancette** [comment]
> @clalancette do we have any official policy about this?

I don't think we have official policy for this, though we are typically conservative about backports for this reason.

**Review by wjwwood — APPROVED** [review]

**Comment by wjwwood** [comment]
I'm fine with reverting but given the turn around on a foxy patch, might be better to fix the bug and push it to all distros, rather than revert, push patch, re apply change, push another patch.

**Comment by fujitatomoya** [comment]
@ivanpauno @wjwwood @clalancette sorry about this to make confusion, i will add the patch so closing this one.

> I don't think we have official policy for this, though we are typically conservative about backports for this reason.

okay, I will at least some period to before the backporting.

> Could you explain the issue in more detail?

will try with the patch.

**Comment by fujitatomoya** [comment]
https://ci.ros2.org/job/ci_windows/17949/ are unrelated, i think we can merge this. CC: @ivanpauno

File: rclpy/rclpy/executors.py [hunk 1]
```diff
@@ -765,11 +763,7 @@ def _spin_once_impl(
         except ConditionReachedException:
             pass
         else:
-            def handler_wrapper(handler):
-                handler()
-                if handler.exception() is not None:
```
Comment (fujitatomoya) [inline rclpy/rclpy/executors.py:770]: This will generates blocking problem https://github.com/ros2/rclpy/issues/1016

**Review by ivanpauno — APPROVED** [review]

**Comment by fujitatomoya** [comment]
@Mergifyio backport foxy galactic humble

**Pull request merged**